src/vivarium/framework/resource/interface.py

```diff
@@ -84,6 +84,8 @@ def add_private_columns(
             A list of resources that the producer requires. A string represents
             a population attribute.
         """
+        if isinstance(resources, str):
+            resources = [resources]
```

Comment:
This is pretty hacky. Since a string is a Sequence, we need to be sure that the thing passed in really is a list.